lib/Pool.js

```diff
@@ -459,29 +465,20 @@ class Pool {
     this._log("force destroying all objects", "info");
 
     const willDie = this._availableObjects;
-    this._availableObjects = [];
-    const todo = willDie.length;
 
     this._removeIdleScheduled = false;
     clearTimeout(this._removeIdleTimer);
 
-    return new Promise(resolve => {
-      if (todo === 0) {
-        return resolve();
-      }
-
-      let resource;
-      let done = 0;
-
-      while ((resource = willDie.shift())) {
-        this.destroy(resource.resource);
-        ++done;
-
-        if (done === todo && resolve) {
-          return resolve();
-        }
-      }
-    });
+    return Promise.all(
+      willDie.map(resource => {
+        return this.destroy(resource.resource);
+      })
+    ).then(
+      () =>
+        (this._availableObjects = this._availableObjects.filter(
```

Comment:
Isn't `willDie == _availableObjects` and `_availableObjects` will be `[]` after this?

Reply:
That was actually my original implementation, but then I realised that `_ensureMinimum()` was called inside `destroy()`, so connections might be being created while destroying the already created ones.

Reply:
Can we ensure this doesn't happen maybe?

Reply:
Do you mean imposing *not* to create new connections when `destroyAllNow()` is called?.

I'd be happy to change that, but looking at the [method description](https://github.com/sequelize/sequelize-pool/blob/9e9bd01c9547c89a97fdd3c96ae93238ac13a59b/lib/Pool.js#L451), that seems to be the specified behavior, so doing so might break some clients.

Looking at [sequelize implementation](https://github.com/sequelize/sequelize/blob/dcf079b25067b5241287c3f6fe82585f4d08b732/lib/dialects/abstract/connection-manager.js#L99) the pool is set to be drained before calling `destroyAllNow()`,and `_ensureMinimum()` doesn't take action in that scenario, so it'd be safe on that side.

What do you think?. I could, for example, set `this._draining` to `true` the method so we'd avoid this scenario and I can safely set `this_availableObjects` to an empty list.